### ISSUE TITLE
Bundle size increased a lot after upgrading from Webpack 2 to Webpack 4

### ISSUE THREAD

**Issue by GozuPro** [webpack/webpack#10383]
I successfully upgraded one project from Webpack 2 to Webpack 4. While doing that, app.js served to users went from 500kb to 2.3Mb which is unacceptable. I believe three shaking isn't configured properly but I could be wrong: https://stackoverflow.com/questions/60198045/webpack-bundle-size-completely-off

<img width="1108" alt="Screen_Shot_2020-02-13_at_10_39_50_AM" src="https://user-images.githubusercontent.com/6925462/74467131-5c86ae00-4e4d-11ea-9dbb-81b84e59a08c.png">

**Comment by sokra** [comment]
Please submit support requests and questions to StackOverflow using the tag [[webpack]](http://stackoverflow.com/tags/webpack). StackOverflow is better suited for this kind of support. The issue tracker is for bug reports and feature discussions. See also [our CONTRIBUTING guidelines](https://github.com/webpack/webpack/blob/master/CONTRIBUTING.md).

**Comment by GozuPro** [comment]
That's exactly what I did : https://stackoverflow.com/questions/60198045/webpack-bundle-size-completely-off. Since, I didn't have answers yet, I also left a note here.

**Comment by ScriptedAlchemy** [comment]
Most likely optimization is improperly configured. Read the docs or throw a repo up on stack overflow. Terser plugin. Minimizer. Module concatenation, providedExports

**Comment by GozuPro** [comment]
Thanks @ScriptedAlchemy. I created a repository since I wasn't progressing the way I wanted: https://github.com/GozuPro/webpack-optimization. I'm not expecting to reduce the file from 2.3 Mb to 200ko yet. If we can get closer to the initial 500ko (Webpack 2.0) without too many changes, that would be great.

**Comment by GozuPro** [comment]
Hello @ScriptedAlchemy, I don't know if you or someone had a chance to take a look at the repository I created? I'm quite stuck! What's frustrating, when I upgraded from 2.0 to Webpack 4.0, I only changes a few things in .babelrc, package.json and webpackconfig. For some reason, the compression is failing, I realized that with Webpack Bundle Analyzer : 

Webpack 2.0 : Gzipped size was 80 Kb
<img width="380" alt="Screen Shot 2020-02-20 at 2 35 23 PM" src="https://user-images.githubusercontent.com/6925462/74992302-88310780-53fc-11ea-9004-723973bec140.png">

Webpack 4.0 : Gzipped size is 676 Kb
<img width="365" alt="Screen Shot 2020-02-20 at 2 35 17 PM" src="https://user-images.githubusercontent.com/6925462/74992327-997a1400-53fc-11ea-9a13-493dab02398e.png">
